src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -373,73 +401,77 @@ public async Task StopAsync(CancellationToken cancellationToken)
 
         public async Task RestartHostAsync(CancellationToken cancellationToken)
         {
-            using (_metricsLogger.LatencyEvent(MetricEventNames.ScriptHostManagerRestartService))
+            if (!ShutdownRequested && Interlocked.Exchange(ref _restartRequested, 1) == 0)
```

Comment:
Had a chat with @yojagad about concerns about possible race conditions with the new approach

Reply:
Addressed this and added additional check in FileMonitoringService to ensure that restart is invoked just once as discussed.